pkg/kubectl/cmd/create.go

```diff
@@ -123,7 +123,7 @@ func (o *CreateOptions) ValidateArgs(cmd *cobra.Command, args []string) error {
 		if len(o.FilenameOptions.Filenames) != 1 {
 			return cmdutil.UsageErrorf(cmd, "--raw can only use a single local file or stdin")
 		}
-		if strings.HasPrefix(o.FilenameOptions.Filenames[0], "http") {
```

Comment:
raw takes a path, which should start with `/`, not `http`

Reply:
This is for filename instead of raw.